### DISCUSSION TITLE
RaspberryMatic als HA-AddOn - Langsame WebUI

### DISCUSSION THREAD

**Discussion by Micha-DU** [OpenCCU/OpenCCU#2503] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,

ich habe vor etwa 2 Jahren einen Raspberry Pi 3B+ zur Installation einer CCU-Umgebung bzw. RaspberryMatic gekauft und eingerichtet, was bisher auch prima funktioniert hat. Vor kurzem bin ich dann auf "HomeAssistant" aufmerksam geworden und bin von dieser Umgebung restlos begeistert. Natürlich war es dann kaum eine Überlegung, HA und das passende RaspberryMatic-AddOn zu installieren.

Das Ganze ging einigermaßen gut von der Hand und funktioniert auch soweit prima, nur bin ich ehrlich gesagt ein bisschen schockiert von der entsprechenden "Performance". Die Anmeldung an RaspberryMatic dauert mitunter 2 bis 3 Minuten, einige Komponenten melden zeitweise, dass sie nicht erreichbar sind (das war vorher nie der Fall) und das ganze wirkt doch irgendwie extrem "zäh" und "wackelig".

Habe ich bei der Installation (in meinem HA sind noch nicht wirklich viele Komponenten installiert) etwas grob falsch gemacht oder kommt der Pi 3B+ mit beidem dann doch mal an seine Grenzen? Ich bin gern bereit, in einen 4er zu investieren, wäre mir aber vorher gern "sicher", dass das die Performance-Probleme auch nachhaltig beseitigt.

Habt Ihr Erfahrungen damit?

LG Micha

**Comment by jens-maus** [comment, marked as answer]
Nun, es kommt natürlich als erstes darauf an was du so alles inzwischen unter HomeAssistant "schönes" machst bzw. wieviel verschiedene Integrationen du da so parallel sonst noch so betreibst. Dann kann es inzwischen mit einem Pi3 schon mitunter etwas knapp werden. Und das ehrlich gesagt weniger bzgl. CPU, sondern mehr anhand des maximal 1GB RAM das dann mitunter schon sehr eng werden kann mit mehrerer Anwendungen innerhalb von HA. Deshalb ist die Default-RaspberryPi Platform für HomeAssistant ja inzwischen ein Pi4.

Auch musst du bedenken das im Falle des RaspberryMatic HA Addons, du die RaspberryMatic quasi innerhalb von HomeAssistant zu einem virtuellen System degradiert hast und auch das natürlich ein gewissen Performanceengpass nun ergeben kann.  

Um das ganze jedoch zu "debuggen" könntest du erst einmal im RaspberyMatic HA-Addon den Port 80/443 nach aussen freigeben damit du z.B. nicht innerhalb von HomeAssistant dann auf die WebUI von RaspberryMatic zugreifst, sondern mal schaust wie sich das System verhält wenn du dann quasi direkt auf die WebUI via durchgeleitetem Port 80/443 zugreifst. Vielleicht liegt es ja nur an der Ingress durchleitung das die WebUI sich "zäh" anfühlt. Ein versuch wäre es wert.

Ansonsten musst du natürlich gerade beim RaspberryPi4 etwas aufpassen bzgl. parallele Nutzung des RPI-RF-MOD / HM-MOD-RPI-PCB Funkmodules, denn wenn man dieses an einem Pi4 betreibt rennt man da mitunter schnell in ein Funkproblem, denn der Pi4 ist dafür gekannt das dieser eine recht große Funkstörwolke um sich herum generiert und dann gerade das RPI-RF-MOD oder das HM-MOD-RPI-PCB HomeMatic Funkmodul dann seine Geräte gar nicht mehr erreichen kann.

Best practice wäre ehrlich gesagt z.B. HomeAssistant auf einem separaten Pi4 zu betreiben und RaspberryMatic dann auf dem Pi3 zu belassen oder noch besser beide Systeme via Proxmox gänzlich z.B auf einem intel NUC System zu virtualisieren und ganz davon abzurücken dafür ein RaspberryPi zu verwenden. Das bringt die meiste Performance und Flexibilität.

**Reply by Micha-DU** [reply]
Hallo Jens,

Danke für das ausführliche Statement... irgendwie hatte ich auch schon so das Gefühl, dass es vielleicht eher die "Durchleitung" ist als die Systemperformance als solche, weil die Ladezeiten nämlich auch total schwanken. Manchmal geht es gefühlt ohne Verzögerung und manchmal meint man, es kommt gar nichts mehr - daher auch meine Frage, ob da ggf. etwas falsch aufgesetzt sein könnte.

Wenn ich die Ports 80 und 443 im HA-AddOn freigebe... wie erreiche ich denn dann die WebUI "direkt"? Einen Eintrag "homematic-raspi" wird das AddOn ja nicht generieren und mit der IP:80 oder :443 funktioniert es auch nicht... das würde ich tatsächlich gern zuerst mal testen.

Wirklich "viel" habe ich in meiner HA-Umgebung noch nicht... 2x iCloud... 1x Miele... 2x KIA connect... HACS.... und das, was so automatisch kommt (Fritz!Box usw.). Die gesamte Tuya-Geschichte ist noch nicht implementiert, weil ich mich noch im Zusammenhang mit cloudfreiem Sprachassistenten usw. einlese.

**Reply by jens-maus** [reply]
> Wenn ich die Ports 80 und 443 im HA-AddOn freigebe... wie erreiche ich denn dann die WebUI "direkt"? Einen Eintrag "homematic-raspi" wird das AddOn ja nicht generieren und mit der IP:80 oder :443 funktioniert es auch nicht... das würde ich tatsächlich gern zuerst mal testen.

Du musst im RaspberryMatic HA-Addon unter Konfiguration -> Network bei Port 80 (WebUI) z.B. den Port 8080 eintragen als den Port unter dem dann das Addon von aussen (d.h. via des HomeAssistant Hosts) erreichbar sein soll. Das ist auch in der Dokumentation (https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant) unter Punkt 7 so zu sehen. Dann kannst du deinen Browser auf sowas wie `http://<homeassistant>:8080/` lenken (statt `<homeassistant` dann IP oder hostname des HomeAssistant Host eintragen) und dann sollte die RaspberryMatic WebUI unter Port 8080 erreichbar sein.

**Reply by Micha-DU** [reply]
Oh sorry... ich bin eigentlich auch ein Verfechter von "RTFM", aber in dem Fall hab ich versagt.

Funktioniert nun prima und was soll ich sagen... die Performance ist genauso schnell wie bei der Einzelinstallation (ohne HA), also wie gewohnt. Vielen lieben Dank!!!

Allerdings bin ich ja so ein Typ, der sich jetzt fragt... was stimmt in meinem Netzwerk nicht, dass die Durchleitung derart "gehakt" hat? Aber das ist hier wahrscheinlich eher off-topic ;-)

**Reply by jens-maus** [reply]
> Allerdings bin ich ja so ein Typ, der sich jetzt fragt... was stimmt in meinem Netzwerk nicht, dass die Durchleitung derart "gehakt" hat? Aber das ist hier wahrscheinlich eher off-topic ;-)

Das wird wohl weniger an deinem Netzwerk selbst liegen, sondern eher daran liegen das bei diesen Ingress-Proxy-Durchleitungen (d.h. das man direkt innerhalb von HA die WebUIs anderer Systeme/Addons einbinden lassen kann) vielleicht diese nicht ganz so performant laufen bzw. es da noch gewisse Bottlenecks gibt. In meinen eigenen kurzen Tests konnte ich da nicht wirklich performane-issues feststellen, kann aber gut sein das bei größeren Installationen ggf. ausgeprägter ist und ich mir das dann mal näher anschauen müsste wo genau das Bottleneck ist. Das es da aber einen gewissen Overhead gibt lässt sich nicht vermeiden und die Umleitung auf Port 8080 ist immer zwangsläufig die performanteste Lösung.

**Reply by Micha-DU** [reply]
Ich kann mir schon vorstellen, dass das sehr "system-spezifisch" ist. Da kann ja durchaus eine "umfangreiche" (speicherhungrige?) Integration das Zünglein an der Waage sein. Ich bin halt eher ein "pingeliger" Typ und habe mit meinen HmIP-BWTHs schon genug Probleme, was die Regelung angeht, da möchte ich nicht auch noch denken müssen, dass die CCU/RaspberryMatic vielleicht "hängt".

**Comment by Baxxy13** [comment]
Ich rate auch dazu RM "woauchimmer" separat zu betreiben. Muss es wirklich all-in-one auf einem SBC sein dann mindestens Pi4B-4GB und mit SSD-Boot. Wobei letzteres auch häufig Probleme macht (Adapter abhängig). Oder halt den Yellow mit SSD und abgesetztem Funkmodul.

Oder halt alternative SBC's (z.B. Odroid) die mehr RAM haben und von SSD oder EMMC booten können.

**Reply by Micha-DU** [reply]
Yellow?

Da ich ohnehin den RF-USB nutze, könnte ich den ja mit einer hochwertig geschirmten Verlängerung aus der "Gefahrenzone" bringen und eine SSD hatte ich ohnehin schon mal überlegt...

**Reply by jens-maus** [reply]
> Yellow?

https://www.home-assistant.io/yellow/

bzw. den kommenden "Green":

https://www.home-assistant.io/green/

> Da ich ohnehin den RF-USB nutze, könnte ich den ja mit einer hochwertig geschirmten Verlängerung aus der "Gefahrenzone" bringen und eine SSD hatte ich ohnehin schon mal überlegt...

Dann wäre bei einem Pi4 nur wichtig den HmIP-RFUSB und auch die SSD auch **nicht** an einen der blauen USB3 Ports anzuschließen, denn das verursacht nur Funkstörungen und dann erreicht der HmIP-RFUSB mitunter seine Geräte auch nicht mehr. Zusätzlich dann vielleicht auch noch sowas wie einen USB Entstörfilter verwenden (https://de.elv.com/elv-usb-entstoerfilter-usb-ef1-komplettbausatz-152745) um den USB Stick auch vor etwaigen Störsignalen noch weiter abzuschirmen.

**Reply by Micha-DU** [reply]
Also die SSD würde doch durchsatztechnisch eigentlich schon an den blauen USD3 gehören...? Ganz ehrlich... ich kann mir nicht vorstellen, dass der Pi4 derart stört, wenn man den RF-USB mit einer Verlängerung aus der Gefahrenzone holt...? Ansonsten würde er ja auch in ein Metallgehäuse gehören?

**Reply by Baxxy13** [reply]
Wenn man um die Funkstörungen des Pi4B weiß bekommt man das auch gehandhabt. 
Genau die USB3-Ports sind die Problemkinder wenn USB3-fähige Geräte (SSD's) angeschlossen sind.
Man kann den Pi4B samt SSD natürlich in ein Metallgehäuse packen und nur das Stromkabel sowie das USB-Kabel für's Funkmodul herausführen.

**Reply by jens-maus** [reply]
> Man kann den Pi4B samt SSD natürlich in ein Metallgehäuse packen und nur das Stromkabel sowie das USB-Kabel für's Funkmodul herausführen.  

Das ist HF technisch jedoch auch suboptimal, denn das USB Kabel gibt dann mitunter eine perfekte Antenne und kann die HF Störungen dann einfach mitunter nach außen führen. Ein USB Entstörfilter und/oder Ferritkern am Kabel würde ich trotz Metallgehäuse weiter als Minimum empfehlen.

**Reply by Baxxy13** [reply]
Stimmt, hatte ich vergessen zu erwähnen. Ist bei meinem "Test-HmIP-RFUSB" beides vorhanden.

**Reply by Micha-DU** [reply]
Ich bin ehrlich gesagt etwas überrascht. Wenn ihr sehen würdet, dass mein Pi3B+ inkl. RF-USB direkt neben meiner Fritz!Box 7530ax liegt, müsstet ihr mich ja eigentlich steinigen? 😂

Also DAS kann ich räumlich gesehen wohl locker vergessen, wenn ich auf einen Pi4 umsteige 🙈

Das genannte Gehäuse sieht echt interessant aus. Und es ist auch immer mal über Kleinanzeigenportale zu bekommen… bleibt die Frage, ob ich einen 4GB oder 8GB nehme…?

**Comment by friedpa** [comment]
Schaue Dir einmal das Argon One Gehäuse mit M.2 Slot an (Achtung SSD darf nicht vom Typ NVME sein). Ich habe damit keine Probleme, allerdings verwende ich das abgesetze HB-RF-ETH Funkmodul.

**Reply by Baxxy13** [reply]
Dann stell das Teil bitte nicht neben eine CCU. 
Die Störstrahlung ist zwar durch den "U-USB-Verbinder" nicht ganz so groß wie bei einem Kabel, aber immer noch "merkbar".
Alles getestet. 😉

**Comment by Micha-DU** [comment]
Hallo zusammen,

heute habe ich dann doch wieder das beschriebene Performance-Problem, auch bei dem direkten Zugriff über den Port 8080. Das Ganze wirkt sehr "zäh" und "wacklig", auch hat der HA teilweise gezickt und das default-dashboard nicht angezeigt.

Bei dem "bisschen" an Integrationen, die ich bisher implementiert habe, macht mich das Ganze doch irgendwie ein bisschen stutzig. Vielleicht ist auch die SD-Karte nicht mehr die fitteste und ich sollte doch auf einen Pi4 mit SSD/m.2 umsteigen?

LG Micha

**Reply by jens-maus** [reply]
Wie gesagt, es wird eher der maximale Speicher von 1GB sein der hier ggf. etwas limitiert bzw. langsamer macht und HomeAssistant ggf. dazu zwingt zu swappen. Deshalb, ja. andere Platform. Ob Pi4 müsste man wie gesagt mit starker Vorsicht wegen funkstörungen probieren/testen. Besser wäre – wie schon erwähnt – einfach gleich ein x86-basiertes IntelNUC System sich zu besorgen, dort Proxmox draufzumachen und dann HomeAssistant und RaspberryMatic als unabhängige VMs darauf zu betreiben. Dann ist man nicht nur alle Kindergartensorgen los, sondern hat auch gleich noch Potential um andere virtuelle Umgebungen darauf zu betreiben...

**Reply by Micha-DU** [reply]
Der Umstieg auf ein wieder neues/anderes System ist für mich eher nicht so die Variante. Ich glaube, da wäre ein Pi4 mit SSD dann doch die bessere Alternative (ja, schlagt mich gern, wenn ich dann irgendwann auch daran verzweilfe und DOCH noch auf sowas wie einen NUC oder so umsteige... wobei ich sogar noch einen Esprimo Q-irgendwas, ich glaube 920, in der Ecke liegen hätte... aber selbst DER wäre doch schon übertrieben).

Als Nutzer eines RF-USB müsste ich doch eigentlich mit einer "guten" Verlängerung in der Lage sein, den Stick aus der Störzone zu bekommen? Nur der Platz in unmittelbarer Nähe zur Fritz!Box sollte dann vielleicht anders gewählt werden ;-)

**Reply by jens-maus** [reply]
Probier es. Gewarnt wurdest du ja jetzt mehrfach :) Oder du wartest bis RaspberryPi5 Support von HomeAssistantOS und RaspberryMatic kommt und dann steigst du darauf um. Aber ob das dieses Jahr noch passiert wage ich aktuell zu bezweifeln.

**Reply by Micha-DU** [reply]
Also irgendwie muss ja denken, dass der Pi4 irgendwie gar nicht zu gebrauchen ist, weil er diese Störfelder produziert? Warum wird das mit dem Pi5 dann besser werden bzw. weiß man das jetzt überhaupt schon? Ein Anbieter eines Pi4 ist auf einen "Yellow" umgestiegen und ich frage mich, warum DAS Ding nicht so streut...?

**Reply by jens-maus** [reply]
> Also irgendwie muss ja denken, dass der Pi4 irgendwie gar nicht zu gebrauchen ist, weil er diese Störfelder produziert? Warum wird das mit dem Pi5 dann besser werden bzw. weiß man das jetzt überhaupt schon?

Klar, da wird an mehreren Fronten heiss dran gearbeitet und es sieht wohl auch danach aus das der Pi5 weniger "stört".

Siehe:
https://homematic-forum.de/forum/viewtopic.php?f=65&t=80283&p=784275#p784267
https://github.com/home-assistant/operating-system/pull/2914

> Ein Anbieter eines Pi4 ist auf einen "Yellow" umgestiegen und ich frage mich, warum DAS Ding nicht so streut...?

Das wird vmtl. daran liegen das der Yellow auf das RaspberryPi Compute Module setzt und nicht einen Standard RaspberryPi4 einsetzt.

**Reply by Micha-DU** [reply]
Ihr wisst hier aber auch einfach immer alles :-) Danke für die vielen Infos.

Also. Ich werde den Schritt mal wagen und einen Pi4B (wenn alles klappt, vorsichtshalber mit 8GB) mit einer SSD in ein Argon One-Gehäuse packen (ich habe jetzt so bei der ersten Durchsicht keine vergleichbare Alternative zu diesem Gehäuse gefunden...?) und dann berichten, ob funktechnisch alles zusammen bricht ;-)
